torch/_dynamo/optimizations/distributed.py

```diff
@@ -236,42 +247,69 @@ def forward(self, *args):
                         return x
 
                 unwrap_singleton_tuple = False
-                for sn in submod.graph.nodes:
+                for sn in real_mod.graph.nodes:
```

Comment:
would it be better to zip the 2 nodes lists and assert one of them matches the other instead of duplicating the traversal separately?

Reply:
Eh, I think this is all just duplicated cause I havent pulled it into a function yet